File: src/AILogistics.Domain/Entities/TrackingEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AILogistics.Domain.Entities
{
    public class TrackingEvent : BaseEntity
    {
        public int ShipmentId { get; set; }
        public ShipmentStatus Status { get; set; }
        public string Location { get; set; }    
        public string Description { get; set; }
        public DateTime EventTime { get; set; }
        public Shipment shipment { get; set; }


    }
}

File: Client/src/types/trackingEvent.ts
import type { ShipmentStatus } from '@/types/shipment';

export interface TrackingEvent {
    id: number;
    shipmentId: number;
    shipmentNumber: string;
    status: ShipmentStatus;
    location: string;
    description: string;
    eventTime: string;
    createdAt: string;
}

export interface CreateTrackingEventRequest {
    shipmentId: number;
    status: ShipmentStatus;
    location: string;
    description: string;
    eventTime: string;
}
